source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -200,6 +200,16 @@ int StreamHandleWrapper::luaHttpCall(lua_State* state) {
   }
 }
 
+int StreamHandleWrapper::luaAddorUpdateCluster(lua_State* state) {
```

Comment:
Added support of Lua. I think this may be more generic as people can pass whatever cluster definition they want in yaml here and the lua will just call `clustermanager`'s  `addOrUpdateCluster`.